platform-services/src/std_platform_services.rs

```diff
@@ -15,14 +17,36 @@
 use super::result;
 
 use getrandom;
+use nix::{
+    time,
+    sys::time::TimeValLike,
+};
 
 /// Fills a buffer, `buffer`, with random bytes sampled from the random number
 /// source provided by the host operating system, as provided by `getrandom`.
-///
-/// This is for use with "freestanding-execution-engine".
-pub fn platform_getrandom(buffer: &mut [u8]) -> result::Result {
+pub fn platform_getrandom(buffer: &mut [u8]) -> result::Result<()> {
     if let Ok(_) = getrandom::getrandom(buffer) {
```

Comment:
Why not use `is_success()` here, rather than a dummy pattern match?

Reply:
This `getrandom()` function belongs to another crate, it doesn't return a `result::Result`